cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -107,17 +110,22 @@ Result<std::shared_ptr<ArrowType>> MakeArrowTimestamp(const LogicalType& logical
   }
 }
 
-Result<std::shared_ptr<ArrowType>> FromByteArray(const LogicalType& logical_type) {
+Result<std::shared_ptr<ArrowType>> FromByteArray(
+    const LogicalType& logical_type, const ArrowReaderProperties& reader_properties) {
   switch (logical_type.type()) {
     case LogicalType::Type::STRING:
       return ::arrow::utf8();
     case LogicalType::Type::DECIMAL:
       return MakeArrowDecimal(logical_type);
     case LogicalType::Type::NONE:
     case LogicalType::Type::ENUM:
-    case LogicalType::Type::JSON:
     case LogicalType::Type::BSON:
       return ::arrow::binary();
+    case LogicalType::Type::JSON:
+      if (reader_properties.known_arrow_extensions_enabled()) {
+        return ::arrow::extension::json(::arrow::utf8());
+      }
+      return ::arrow::binary();
```

Comment:
I'm a bit curious here, would this a binary or a string?

Reply:
I think, as per @progger-dev's comment in tests:
>  // When the original Arrow schema isn't stored and Arrow extensions are disabled,
>  // LogicalType::JSON is read as Binary.

Reply:
Should I add a comment?

Reply:
Let's better doing this, thanks! Though I don't so like the idea to use binary, I think `utf8` is better, in parquet-spec, json is "UTF8 column": https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L154

I also want to hear what you think @pitrou @wgtmac

Reply:
Yeah, string seems to make more sense. Changed https://github.com/apache/arrow/pull/13901/commits/23b325ef0afdf4796c9e284e84772f9a3b98a26d, please see if this looks better @mapleFU.